src/main/scala/com/github/unchama/seichiassist/subsystems/tradesystems/subsystems/gttosiina/bukkit/traderules/BukkitTrade.scala

```diff
@@ -0,0 +1,58 @@
+package com.github.unchama.seichiassist.subsystems.tradesystems.subsystems.gttosiina.bukkit.traderules
+
+import cats.effect.ConcurrentEffect
+import cats.effect.ConcurrentEffect.ops.toAllConcurrentEffectOps
+import com.github.unchama.seichiassist.SeichiAssist
+import com.github.unchama.seichiassist.subsystems.gacha.GachaAPI
+import com.github.unchama.seichiassist.subsystems.gacha.domain.CanBeSignedAsGachaPrize
+import com.github.unchama.seichiassist.subsystems.gacha.domain.GachaRarity.GachaRarity
+import com.github.unchama.seichiassist.subsystems.gacha.domain.GachaRarity.GachaRarity.Gigantic
+import com.github.unchama.seichiassist.subsystems.gacha.subsystems.gachaprizefactory.bukkit.StaticGachaPrizeFactory
+import com.github.unchama.seichiassist.subsystems.tradesystems.domain.{
+  TradeResult,
+  TradeRule,
+  TradeSuccessResult
+}
+import org.bukkit.entity.Player
+import org.bukkit.inventory.ItemStack
+
+object BukkitTrade {
+
+  import cats.implicits._
+
+  def apply[F[_]: ConcurrentEffect](name: String)(
+    implicit gachaAPI: GachaAPI[F, ItemStack, Player],
+    canBeSignedAsGachaPrize: CanBeSignedAsGachaPrize[ItemStack]
+  ): TradeRule[ItemStack] =
+    (contents: List[ItemStack]) => {
+      val eff = for {
+        gachaList <- gachaAPI.list
+      } yield {
+        val giganticItemStacks = gachaList // TODO GTアイテムかどうかを確率に依存すべきではない
```

Comment:
```suggestion
        val giganticItemStacks = gachaList
```

すでにラップすることによって抽象化されているのでこのTODOコメントは不要